### PR TITLE
Milestone2

### PR DESCRIPTION
**Pull request by gasb150**
For this milestone, we made the next steps:

- Prepare the directory structure
- Implement the App component
- Implement the BooksList component
- Implement the Book component
- Implement the BooksForm component
- Define the Redux actions and reducers
- Initialize the Redux store

### CODE REVIEW

**Review by teekaytech — CHANGES_REQUESTED** [review]
### Status: Invalid Code Review Request :x: 

__Hi team, I commend you for the great effort put into this milestone. However, I have to mark the project as invalid due to the following reasons:__

- [ ] One of your linter checks is not passing successfully.
- For this project to be considered for review, you need to ensure that all linter checks are running and passing successfully. Kindly fix all linter errors and request another review when you're ready.

**_This Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready!_**


* * *
**Please, do not open a new Pull Request for subsequent reviews. You should use the same Pull Request submitted for the first review unless it is requested otherwise.**

**Always post your inquiries and questions on this PR thread as a comment. You can mention me (@teekaytech) so I can see your comments and respond.**

* * * 

**_You can reach out to me on Twitter (@ola_lere), Github(@teekaytech), or slack (@taofeekolalere )_**

Happy Coding :man_technologist:

File: src/components/BookList.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+);
+const mapStateToProps = (state) => ({ books: state.books });
+
+BooksList.propTypes = {
+  books: PropTypes.objectOf(PropTypes.array).isRequired,
+};
+
+// BooksList.defaultProps = {
+//   books: [],
+// };
```
Comment (Temmarie) [inline src/components/BookList.js]: - [ ] _[REQUIRED]_ Kindly remove this commented out code.

File: src/components/BooksForm.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  <div>
+    <form>
+      <label htmlFor="title">
+        Title
+        <input id="title" name="title" type="text" />
+      </label>
+      <label htmlFor="options">
+        Title
+        <select name="options" id="options">
+          {booksCategories.map((category) => (
```
Comment (Temmarie) [inline src/components/BooksForm.js]: - [ ] _[REQUIRED]_ This file should be placed in the `src/containers` directory.
- [ ] _[REQUIRED]_ It is the `BookForm` component not `BooksForm` as you have in your code.

File: src/components/BookList.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+const BooksList = ({ books }) => (
+
+  <table>
+    <thead>
+      <tr>
+        <th>Book Id</th>
+        <th>Title</th>
+        <th>Category</th>
+      </tr>
+      <tr>
```
Comment (Temmarie) [inline src/components/BookList.js]: - [ ] _[REQUIRED]_ On running this project, I get a `TypeError: Cannot read property 'map' of undefined`. Kindly fix this.

**Review by Temmarie — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED! :recycle:

### Hi @gasb150 and @edxco    ! :wave: :wave:

#### You have done really well in implementing the requirements.
- You used proper GitHubFlow,
- Your README and PR message are descriptive. 👍🏾 

 However, there are changes required for you to make in order for this project to be approved. 
- - -

### **Recommended** ♻️

- [ ] Your README could be descriptive enough. You currently do not have the authors' contact on the README. Consider adding this.
- [ ] On running your code, I get a TypeError. See my comment below and kindly fix this.


_See the inline comments below for more required changes_

- - - 

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

File: src/components/BooksForm.js (hunk 2 above)
Comment (teekaytech) [inline src/components/BooksForm.js]: **Required**
- [ ] As required by the last reviewer, you need to create a folder called `containers`, then move the `BookForm.js` and `BookList.js` components into it.
- [ ] Also, it is good to follow instructions as laid down by the project specification. Kindly rename the `BooksForm` to `BookForm` as specified in the spec.

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

_You've done really well so far. Some of the changes required by the last reviewer have been implemented._ :+1: 

__However, some of the required changes still have to be taken care of before moving on to the next milestone. Please check my comments below and fix them as appropriate.__

### **Required Changes**

- Check the comments under the review.

### **Optional Changes**
_Every comment with the **OPTIONAL** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also:
- Check the browser console. You're having some errors/warnings that need to be taken care of :blush:
- Improve the Readme file. Information like project authors' details is still missing.


* * *
**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise.**

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@teekaytech) so I can see your comments and respond._

_**Please make the required changes and submit another review request when you are done.**_

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Cheers and happy coding :smiley:__

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi @edxco, congratulations on this milestone :smile: The BookForm component has been renamed as required :slightly_smiling_face: It has also been moved into the "container" directory along with the BookList component:+1: 

I hereby approve the project :+1: 
Kindly go ahead and merge into the development
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**